catalogs/catalog-common/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonPropertiesUtils.java

```diff
@@ -0,0 +1,95 @@
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.gravitino.storage.OSSProperties;
+import org.apache.gravitino.storage.S3Properties;
+
+public class PaimonPropertiesUtils {
+
+  // Map that maintains the mapping of keys in Gravitino to that in Paimon, for example, users
+  // will only need to set the configuration 'catalog-backend' in Gravitino and Gravitino will
+  // change it to `catalogType` automatically and pass it to Paimon.
+  public static final Map<String, String> GRAVITINO_CONFIG_TO_PAIMON;
+
+  static {
+    Map<String, String> map = new HashMap();
+    map.put(PaimonConstants.CATALOG_BACKEND, PaimonConstants.CATALOG_BACKEND);
+    map.put(PaimonConstants.GRAVITINO_JDBC_DRIVER, PaimonConstants.GRAVITINO_JDBC_DRIVER);
+    map.put(PaimonConstants.GRAVITINO_JDBC_USER, PaimonConstants.PAIMON_JDBC_USER);
+    map.put(PaimonConstants.GRAVITINO_JDBC_PASSWORD, PaimonConstants.PAIMON_JDBC_PASSWORD);
+    map.put(PaimonConstants.URI, PaimonConstants.URI);
+    map.put(PaimonConstants.WAREHOUSE, PaimonConstants.WAREHOUSE);
+    map.put(PaimonConstants.CATALOG_BACKEND_NAME, PaimonConstants.CATALOG_BACKEND_NAME);
+    // S3
+    map.put(S3Properties.GRAVITINO_S3_ENDPOINT, PaimonConstants.S3_ENDPOINT);
+    map.put(S3Properties.GRAVITINO_S3_ACCESS_KEY_ID, PaimonConstants.S3_ACCESS_KEY);
+    map.put(S3Properties.GRAVITINO_S3_SECRET_ACCESS_KEY, PaimonConstants.S3_SECRET_KEY);
+    // OSS
+    map.put(OSSProperties.GRAVITINO_OSS_ENDPOINT, PaimonConstants.OSS_ENDPOINT);
+    map.put(OSSProperties.GRAVITINO_OSS_ACCESS_KEY_ID, PaimonConstants.OSS_ACCESS_KEY);
+    map.put(OSSProperties.GRAVITINO_OSS_ACCESS_KEY_SECRET, PaimonConstants.OSS_SECRET_KEY);
+    GRAVITINO_CONFIG_TO_PAIMON = Collections.unmodifiableMap(map);
+  }
+
+  /**
+   * Converts Gravitino properties to Paimon catalog properties, the common transform logic shared
+   * by Spark connector, Gravitino Paimon catalog.
+   *
+   * @param gravitinoProperties a map of Gravitino configuration properties.
+   * @return a map containing Paimon catalog properties.
+   */
+  public static Map<String, String> toPaimonCatalogProperties(
+      Map<String, String> gravitinoProperties) {
+    Map<String, String> paimonProperties = new HashMap<>();
+    gravitinoProperties.forEach(
+        (key, value) -> {
+          if (GRAVITINO_CONFIG_TO_PAIMON.containsKey(key)) {
+            paimonProperties.put(GRAVITINO_CONFIG_TO_PAIMON.get(key), value);
+          }
+        });
+    return paimonProperties;
+  }
+
+  /**
+   * Get catalog backend name from Gravitino catalog properties.
+   *
+   * @param catalogProperties a map of Gravitino catalog properties.
+   * @return catalog backend name.
+   */
+  public static String getCatalogBackendName(Map<String, String> catalogProperties) {
+    String backendName = catalogProperties.get(PaimonConstants.CATALOG_BACKEND_NAME);
```

Comment:
Is `catalog-backend-name` is a new added property for Paimon catalog? could you add related document?

Reply:
sure

Reply:
updated.